unicore/distribute/scripts.py

```diff
@@ -0,0 +1,86 @@
+import os
+import argparse
+
+from elasticgit.storage import StorageManager
+
+from pyramid.paster import bootstrap
+from pyramid.request import Request
+
+from unicore.distribute.utils import get_repositories
+from unicore.webhooks.events import WebhookEvent
+
+
+class PollRepositories(object):
+
+    # This gets set at run time or is overriden for tests
+    notify = None
+
+    def run(self, repo_dir, ini_file, base_url):
+        request = Request.blank('/', base_url=base_url)
+        env = bootstrap(ini_file, request=request)
+        self.notify = env['registry'].notify
+        for repo in get_repositories(repo_dir):
+            self.pull_repo(env, repo)
+
+        # http://docs.pylonsproject.org/projects/pyramid/en/1.3-branch/narr/commandline.html#cleanup # noqa
+        env['closer']()
+
+    def pull_repo(self, env, repo):
+        sm = StorageManager(repo)
+        remote = repo.remotes[0]
```

Comment:
At risk of being a pain, is there a way to get the upstream for the active branch explicitly?